### ISSUE TITLE
WinUI - TemplateBinding not working for Commands

### ISSUE THREAD

**Issue by DRAirey1** [microsoft/microsoft-ui-xaml#7003]
### Describe the bug

TemplateBinding to an ICommand property doesn't work in WinUI. However, explicitly binding using the RelativeSource does.
[BgLabelControlApp.zip](https://github.com/microsoft/microsoft-ui-xaml/files/8524710/BgLabelControlApp.zip)
This doesn't work:

```
<ResourceDictionary xmlns:local="using:ClassLibrary1"
					xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
	<Style TargetType="local:BgLabelControl">
		<Setter Property="Template">
			<Setter.Value>
				<ControlTemplate TargetType="local:BgLabelControl">
					<Grid Background="{TemplateBinding Background}"
						  Height="100"
						  Width="100">
						<Button Content="{TemplateBinding Label}"
								HorizontalAlignment="Center"
                                Command="{TemplateBinding GoBack}"
					</Grid>
				</ControlTemplate>
			</Setter.Value>
		</Setter>
	</Style>
</ResourceDictionary>

```
This works:

```
<ResourceDictionary xmlns:local="using:ClassLibrary1"
					xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
	<Style TargetType="local:BgLabelControl">
		<Setter Property="Template">
			<Setter.Value>
				<ControlTemplate TargetType="local:BgLabelControl">
					<Grid Background="{TemplateBinding Background}"
						  Height="100"
						  Width="100">
						<Button Content="{TemplateBinding Label}"
								HorizontalAlignment="Center"
                                Command="{Binding RelativeSource={RelativeSource TemplatedParent}, Path=GoBack, Mode=OneWay}"
								VerticalAlignment="Center"/>
					</Grid>
				</ControlTemplate>
			</Setter.Value>
		</Setter>
	</Style>
</ResourceDictionary>
```

### Steps to reproduce the bug

1. Expand the attached zip file.
2. Run the application. The second button doesn't work
3. Replace the TemplateBinding with "{Binding RelativeSource={RelativeSource TemplatedParent}, Path=GoBack, Mode=OneWay}"
4. Run the application. The second button works.

### Expected behavior

One expects TemplateBinding to bind to the templated parent.

### Windows app type

- [ ] UWP
- [ ] Win32

### Device form factor

Desktop

### Windows version

Windows 11 (21H2): Build 22000

**Comment by MikeHillberg** [comment]
{TemplateBinding} only supports binding to a property that's backed by a DependencyProperty, which could be causing this. {Binding} uses reflection and generated code-behind to bind to any WinRT or C# property.

**Comment by DRAirey1** [comment]
Did you look at the code? What makes you think it's not binding to a DependencyProperty?

**Comment by kmgallahan** [comment]
Dupe of #4546 ?

**Comment by DRAirey1** [comment]
Yeah, it's likely. The notes say that someone's having trouble with the repro, so the ZIP file here is an easy way to validate that you have a real bug.

**Comment by MikeHillberg** [comment]
> Did you look at the code? What makes you think it's not binding to a DependencyProperty?

Sorry, I was looking at the code in the Issue and didn't notice the zip file link. Thanks for the repro.

**Comment by kmgallahan** [comment]
May also be related to #2508 in that x:Bind behevior in templates & resource dictionaries is finicky, failing there when:

> x:Bind to a property outside the data template works fine ✅
> x:Bind to that same property but with function binding fails to build ❌
> x:Bind to that same property from a visual state setter crashes at runtime ❌
> x:Bind to that same property with a converter crashes at runtime ❌

**Comment by DRAirey1** [comment]
> May also be related to #2508 in that x:Bind behavior in templates & resource dictionaries is finicky, failing there when:
No. x:Bind is done in code behind. This is not code-behind related code.

**Comment by Panda-Sharp** [comment]
Just started to play with WinUI, I started with 1.2, and I just hit this issue #4546

**Comment by andreaslennartz** [comment]
Just stumbled across this exact same issue when creating a very simple CustomControl. 

Can confirm that this is still a thing in 1. Template binding works properly for all other properties, but for some reason it doesn't work with commands. The mentioned workaround (`{Binding RelativeSource={RelativeSource TemplatedParent}, Path=GoBack, Mode=OneWay}`) works. 

The solution is simple, but took me a while to find this - and to me this a very annoying bug.  Should have a higher priority. Hope this can get fixed soon!

**Comment by Soap-141** [comment]
Any update on this?

**Comment by niels9001** [comment]
Closing as a duplicate of #4546.

Both report a TemplateBinding to an ICommand dependency property inside a control template failing, while an explicit TemplatedParent binding works. Same property type, template scenario, and workaround.

Let's consolidate discussion in #4546. If you believe this is not a duplicate, please feel free to reopen — thanks!
